banking-app/test_docker.sh

```diff
@@ -0,0 +1,231 @@
+        "cert": "${member1_cert}\n",
+        "encryption_pub_key": "${member1_encryption_pub_key}\n"
+      }
+    },
+    {
+      "name": "set_user",
+      "args": {
+        "cert": "${user0_cert}\n"
+      }
+    },
+    {
+      "name": "set_user",
+      "args": {
+        "cert": "${user1_cert}\n"
+      }
+    },
+    {
+      "name": "transition_service_to_open",
+      "args": {
+        "next_service_identity": "${service_cert}\n"
+      }
+    }
+  ]
+}
+JSON
+}
+
+##############################################
+# Discover docker configuration
+##############################################
+containerId=$(docker ps -f ancestor=banking-app:virtual -q)
+docker cp "$containerId:/app/service_cert.pem" ./workspace/docker_certificates
+dockerIPAddress=$(docker inspect --format '{{ .NetworkSettings.IPAddress }}' $containerId)
+server="https://${dockerIPAddress}:8080"
+
+##############################################
+# Create all the certs in a well known directory
+##############################################
+cd workspace/docker_certificates
+
+##############################################
+# Activate member 0
+##############################################
+echo "Getting list of members..."
+curl $server/gov/members \
+    --cacert service_cert.pem \
+    | jq
+
+curl "${server}/gov/ack/update_state_digest" \
+    -X POST \
+    --cacert service_cert.pem \
+    --key member0_privk.pem \
+    --cert member0_cert.pem \
+    --silent | jq > activation.json
+
+echo "Show digest"
+cat activation.json
+
+/opt/ccf/bin/scurl.sh "${server}/gov/ack" \
+    --cacert service_cert.pem \
+    --signing-key member0_privk.pem \
+    --signing-cert member0_cert.pem \
+    --header "Content-Type: application/json" \
+    --data-binary @activation.json
+
+echo "Getting list of members..."
+curl ${server}/gov/members \
+    --cacert service_cert.pem \
+    | jq
+
+##############################################
+# Generate Proposals
+##############################################
+create_test_network_proposal
+
+##############################################
+# Open Network
+##############################################
+echo "Open the network"
+network_proposal_out=$(/opt/ccf/bin/scurl.sh "${server}/gov/proposals" \
+    --cacert service_cert.pem \
+    --signing-key member0_privk.pem \
+    --signing-cert member0_cert.pem \
+    --data-binary @network_open_proposal.json \
+    -H "content-type: application/json")
+echo ${network_proposal_out} | jq
+network_proposal_out_id=$( jq -r  '.proposal_id' <<< "${network_proposal_out}" )
+
+echo "Network Proposal ID: $network_proposal_out_id"
+/opt/ccf/bin/scurl.sh "${server}/gov/proposals/$network_proposal_out_id/ballots" \
+    --cacert service_cert.pem \
+    --signing-key member0_privk.pem \
+    --signing-cert member0_cert.pem \
+    --data-binary @vote_accept.json \
+    -H "content-type: application/json" | jq
+
+##############################################
+# Propose users and application
```

Comment:
nit:

You are proposing your users on the same step in which you open the network. not with the application

Reply:
Yes, this is because we use the set_app_js that is generated when we build the actual app - I have added a new story so that we can make this all a bit better: - #58